platforms/hermes-javascript/README.md

```diff
@@ -0,0 +1,654 @@
+
+// You can also return an object that will be used for
+// the 'continue_session' or 'end_session' parameters.
+
+dialog.flow('intentName', (message, flow) => {
+
+    // Chain flow actions (continue / end)…
+
+    return {
+        text: 'intentName recognized!'
+    }
+})
+
+// If you need to perform asynchronous calculations
+// Just return a promise and the flow actions will
+// be performed afterwards.
+
+dialog.flow('intentName', async (message, flow) => {
+    const json = await fetch('something').then(res => res.json())
+
+    // Chain flow actions (continue / end)…
+
+    return 'Fetched some stuff!'
+})
+```
+
+#### flows([{ intent, action }])
+
+Same as `flow()`, but with multiple starting intents.
+
+Useful when designing speech patterns with loops `((intentOne or intentTwo) -> intentTwo -> intentOne -> intentTwo -> ...`,
+so that the starting intents callbacks will **not** get called multiple times if a session is already in progress.
+
+```js
+const intents = [
+    {
+        intent: 'intentOne',
+        action: (msg, flow) => { /* ... */ }
+    },
+    {
+        intent: 'intentTwo',
+        action: (msg, flow) => { /* ... */ }
+    }
+]
+dialog.flows(intents)
+```
+
+#### flow->continue(intentName, action)
+
+Subscribes to an intent for the next dialog step.
+
+```js
+dialog.flow('intentName', async (message, flow) => {
+
+    flow.continue('otherIntent', (message, flow) => {
+        /* ... */
+    })
+
+    flow.continue('andAnotherIntent', (message, flow) => {
+        /* ... */
+    })
+
+    return 'Continue with either one of these 2 intents.'
+})
+```
+
+#### flow->notRecognized(action)
+
+Add a callback that is going to be executed if the intents failed to be recognized.
+
+```js
+dialog.flow('intentName', async (message, flow) => {
+
+    /* Add continuations here ... */
+
+    flow.notRecognized((message, flow) => {
+        /* ... */
+    })
+
+    return 'If the dialog failed to understand the intents, notRecognized callback will be called.'
+})
+```
+
+#### flow->end()
+
+Ends the dialog flow.
+
+```js
+dialog.flow('intentName', async (message, flow) => {
+    flow.end()
+    return 'Dialog ended.'
+})
+```
+
+### Injection Api Subset
+
+Vocabulary injection for the speech recognition.
+
+#### Events available for publishing
+
+- **injection_request**
+
+Requests custom payload to be injected.
+
+```js
+injection.publish('injection_request', {
+    id: /* string */,
+    cross_language: /* string */,
+    // An object having string keys mapped with an array of string entries
+    lexicon: {
+        films : [
+            'The Wolf of Wall Street',
+            'The Lord of the Rings'
+        ]
+    },
+    // An array of operations objects
+    operations: [
+        {
+            // Enumeration: 1 or 2
+            // 1 is 'add', 2 is 'add_from_vanilla'
+            // see documentation here: https://docs.snips.ai/guides/advanced-configuration/dynamic-vocabulary#3-inject-entity-values
+            kind: 1,
```

Comment:
you don't have any way to define a constant in js ? would be nice to have some sort of `add=1` `add_from_vanilla=2` somewhere

Reply:
good point I exposed enums for the lib consumers now